xdsl/irdl/declarative_assembly_format.py

```diff
@@ -1200,29 +1211,16 @@ class OptionalAttributeVariable(AttributeVariable):
     The directive will request a space to be printed after.
     """
 
-    def parse(self, parser: Parser, state: ParsingState) -> bool:
+    def parse_attr(self, parser: Parser) -> Attribute | None:
         # Only qualified optional attributes and symbol names can be optionally
         # parsed currently.
         # Other attributes are parsed as required attributes.
         if self.is_symbol_name:
-            attr = parser.parse_optional_symbol_name()
+            return parser.parse_optional_symbol_name()
         elif self.unique_base is None:
-            attr = parser.parse_optional_attribute()
-        else:
-            return super().parse(parser, state)
-        if attr is None:
-            return False
-        if self.is_property:
-            state.properties[self.name] = attr
+            return parser.parse_optional_attribute()
         else:
-            state.attributes[self.name] = attr
-        return True
-
-    def is_present(self, op: IRDLOperation) -> bool:
```

Comment:
why remove this?

Reply:
an error, thanks for catching this

Reply:
Oh I see you've changed the superclass definition